pylint/checkers/refactoring/refactoring_checker.py

```diff
@@ -2054,17 +2054,21 @@ def _is_function_def_never_returning(
         Returns:
             bool: True if the function never returns, False otherwise.
         """
-        if isinstance(node, (nodes.FunctionDef, astroid.BoundMethod)) and node.returns:
-            return (
+        try:
+            if node.qname() in self._never_returning_functions:
+                return True
+        except (TypeError, AttributeError):
+            pass
+        return (
+            isinstance(node, (nodes.FunctionDef, astroid.BoundMethod))
+            and node.returns
+            and (
                 isinstance(node.returns, nodes.Attribute)
-                and node.returns.attrname == "NoReturn"
+                and node.returns.attrname in {"NoReturn", "Never"}
                 or isinstance(node.returns, nodes.Name)
-                and node.returns.name == "NoReturn"
+                and node.returns.name in {"NoReturn", "Never"}
             )
-        try:
-            return node.qname() in self._never_returning_functions
-        except (TypeError, AttributeError):
-            return False
+        )
```

Comment:
In #9714 the strategy is to start relying on `utils.is_terminating_func()` instead of duplicating this logic in several places. Suggest exploring that here.